packages/api/src/middleware/validation/checkAnimalOrBatch.js

```diff
@@ -171,78 +171,71 @@ const checkAnimalBreed = async (
+  if (someExists(breedKeyOptions, animalOrBatch) || someExists(typeKeyOptions, animalOrBatch)) {
+    // Check only one breed option is truthy
+    if (someTruthy([default_breed_id, custom_breed_id, breed_name])) {
+      checkExactlyOneIsProvided(
+        [default_breed_id, custom_breed_id, breed_name],
+        'default_breed_id, custom_breed_id, or breed_name',
       );
     }
-    // Check if custom breed or custom type is present
-    if (
-      (someExists(breedKeyOptions, animalOrBatch) && custom_breed_id && !type_name) ||
-      (someExists(typeKeyOptions, animalOrBatch) &&
-        (default_type_id || custom_type_id) &&
-        !breed_name)
-    ) {
-      let customBreed;
-      // Find customBreed if exists
-      if (someExists(breedKeyOptions, animalOrBatch) && custom_breed_id) {
-        checkIdIsNumber(custom_breed_id);
-        customBreed = await CustomAnimalBreedModel.query()
-          .whereNotDeleted()
-          .findById(custom_breed_id);
-        if (!customBreed) {
-          throw customError('Custom breed does not exist');
-        }
-      } else if (preexistingAnimalOrBatch?.custom_breed_id) {
-        checkIdIsNumber(preexistingAnimalOrBatch?.custom_breed_id);
-        customBreed = await CustomAnimalBreedModel.query()
-          .whereNotDeleted()
-          .findById(preexistingAnimalOrBatch.custom_breed_id);
-        if (!customBreed) {
-          // This should not be possible
-          throw customError('Custom breed does not exist');
-        }
-      }
-      // Check custom breed if exists
-      if (customBreed) {
-        await checkRecordBelongsToFarm(customBreed, farm_id, 'custom breed');
-        checkCustomBreedMatchesType(
-          animalOrBatch,
+    // Overwrite all others with null in db if editing, post does not accept nulled values in oneOf schemas
+    if (!creating && someExists(breedKeyOptions, animalOrBatch)) {
+      setFalsyValuesToNull(breedKeyOptions, animalOrBatch);
+    }
+
+    const isNotNullingAllBreedOptions = !(
+      someExists(breedKeyOptions, animalOrBatch) &&
+      !someTruthy([default_breed_id, custom_breed_id, breed_name])
+    );
+    const isCreatingWithBreed =
+      creating && someTruthy([default_breed_id, custom_breed_id, breed_name]);
+    // Do checks on breed unless removing breed from existing record or
+    if (isNotNullingAllBreedOptions || isCreatingWithBreed) {
+      // Check if default breed or default type is present
+      if (default_breed_id || default_type_id) {
+        await checkDefaultBreedMatchesType(
           preexistingAnimalOrBatch,
-          customBreed,
+          default_breed_id,
           default_type_id,
-          custom_type_id,
         );
       }
+      // Check if custom breed or new type is present
+      // Skip checks in some cases where new custom type or breed is added
+      if ((custom_breed_id && !type_name) || ((default_type_id || custom_type_id) && !breed_name)) {
```

Comment:
The combination of breed ID and type_name is wrong... isn't it already handled earlier? I'm wondering if it can be `if (custom_breed_id || ((default_type_id || custom_type_id) && !breed_name))`

Reply:
So this is one of my failures of having enough time to structure it how I wanted. It is actually currently handled later.... and you will see that if you change the logic to your suggestion, you are right it will be caught here!.. However it is not the desired error message. 

What I would like to do is chunk up and move both `checkAndAddCustomTypesOrBreeds()` & `checkCustomTypeAndBreedConflicts()` inside the `checkType()` and `checkBreed()` functions.

I did not feel I could do this a the time because:
- This had already taken me an uncomfortably long amount of time to understand and adapt the logic.
- These functions were mostly pre-existing and I did not want to change the order of other work unless necessary, one of my personal goals is to understand when I should make improvements vs leaving the properly functioning logic as is. I think I have received some peer feedback related to this. 

So I chose to name and reuse existing logic but I really did not add or re-order much. I only added a little bit of new logic for comparing existing record and updates and tried to keep the order of logic the same ..as best I could to understand it there are exceptions.

Reply:
Understood, thank you for the detail!